### PR TITLE
Fix VerifyEvents to handle amounts exceeding int64 range

### LINKED ISSUE #5929: Fix VerifyEvents to support amounts exceeding int64 range
### What did you do?

  The VerifyEvents function in the token transfer processor ([verify_events.go#L161](https://github.com/stellar/go-stellar-sdk/blob/main/processors/token_transfer/verify_events.go#L161))
  expects the amount field to be int64. However, SAC token balances can exceed the
  int64 limit through cumulative mints.

  Observed behavior:
  A SAC token's balance grew via mints and was then fully burned with an amount
  (18446947143889701584) exceeding the int64 max. This caused VerifyEvents to fail.

  Evidence:
  - Contract: CBMFDIRY5OKI4JJURXC4SMEQPWB4UUADIADJK4NA6CYBNOYK4W4TMLLF
  - Key: 98a39d51d4d4be729cb01de4c59d489192ce85bbb412b0ff87d6a73cf215cdf6
  - Ledger where balance exceeded int64: 61456768
  - Ledger where full balance was burned: 61457010

### What did you expect to see?

VerifyEvents should not fail for larger amount and supports  amounts larger than int64. Probably we need to find limits.

### What did you see instead?
VerifyEvents failed for `18446947143889701584`

### LINKED ISSUE #5931: ingest: LedgerTransaction.NewMaxFee() truncates int64 fee-bump fee to uint32, causing silent data corruption
### What version are you using?

```
go list -m github.com/stellar/go-stellar-sdk
```

Bug is present in current `main` branch at `ingest/ledger_transaction.go:739`.

---

### What did you do?

Called `LedgerTransaction.NewMaxFee()` on a fee-bump transaction where the outer fee (`FeeBumpTransaction.Fee`) exceeded 4,294,967,295 stroops (~429.5 XLM):

```go
tx, _ := reader.Read()
fee, ok := tx.NewMaxFee()
// fee is silently truncated for any fee-bump tx with fee > 429.5 XLM
```

Fee-bump transactions with fees exceeding this threshold are realistic for resource-intensive Soroban smart contract executions or during network congestion. The Stellar XDR spec deliberately defines `FeeBumpTransaction.Fee` as `Int64` to accommodate values above the `Uint32` range.

---

### What did you expect to see?

`NewMaxFee()` should return the full correct `int64` value of the fee-bump outer fee, consistent with:
- The XDR spec: `FeeBumpTransaction.Fee` is `Int64`
- The underlying accessor: `Envelope.FeeBumpFee()` already returns `int64`

For example, a fee of **5,000,000,000 stroops** (500 XLM) should be returned as `5000000000`.

---

### What did you see instead?

`NewMaxFee()` silently truncates the `int64` fee to `uint32`, discarding the upper 32 bits and returning a completely wrong value with no error.

**Vulnerable code — `ingest/ledger_transaction.go:739`:**
```go
func (t *LedgerTransaction) NewMaxFee() (uint32, bool) {
    if !t.Envelope.IsFeeBump() {
        return 0, false
    }
    return uint32(t.Envelope.FeeBumpFee()), true  // ← BUG: int64 truncated to uint32
}
```

**Example of data corruption:** A fee of 5,000,000,000 stroops becomes **705,032,704** (~70.5 XLM instead of ~500 XLM). The math: `5,000,000,000 mod 4,294,967,296 = 705,032,704`.

**Root cause:** `MaxFee()` on a regular inner transaction correctly returns `uint32` because the inner transaction fee uses XDR `Uint32`. `NewMaxFee()` was modelled after it without accounting for the fact that the fee-bump outer fee uses XDR `Int64`. The wrong type and cast were copied without adjustment.

**Proposed fix:**
```go
func (t *LedgerTransaction) NewMaxFee() (int64, bool) {
    if !t.Envelope.IsFeeBump() {
        return 0, false
    }
    return t.Envelope.FeeBumpFee(), true
}
```

**This is a breaking change** to the public function signature. All callers must update their type handling from `uint32` to `int64`.

**Known affected downstream consumer:** `stellar/stellar-etl` has the same truncation bug independently at `internal/transform/transaction.go:294` and `internal/transform/schema.go:63` — that fix should be coordinated with this one.

**Impact:** Any application using `LedgerTransaction.NewMaxFee()` from the ingest SDK silently receives an incorrect fee value for any fee-bump transaction with a fee exceeding ~429.5 XLM. This affects all downstream analytics, reconciliation, or compliance systems built on this function.

### PR DESCRIPTION
**Pull request by karthikiyer56**
This PR fixes 2 issues:

1/
Replace int64 balance tracking with *big.Int in `findBalanceDeltasFromEvents` and all related functions. This fixes the unrecovered panic in `MustParseInt64Raw` when processing SAC events with i128 amounts that exceed int64 range (e.g., cumulative mints producing balances >`2^63-1`).

`findBalanceDeltasFromEvents` now returns an error instead of panicking on unparseable amounts. No exported API signatures changed.

Fixes #5929

---

2/
Fix `NewMaxFee()` which was truncating the int64 fee-bump fee to uint32, silently corrupting any fee-bump fee above ~429.5 XLM. 
Return type changed from uint32 to int64 (**breaking change**)

Fixes #5931

### CODE REVIEW

File: processors/token_transfer/verify_events.go [hunk 1]
```diff
@@ -18,18 +18,22 @@ type balanceKey struct {
...
-	if m[key] == 0 {
-		delete(m, key)
+	if existing, ok := m[key]; ok {
+		existing.Add(existing, delta)
+		if existing.Sign() == 0 {
+			delete(m, key)
+		}
+	} else {
+		m[key] = new(big.Int).Set(delta)
 	}
```
[the remark this answers is not in the document]
Comment (karthikiyer56) [reply processors/token_transfer/verify_events.go:36]: Good catch. Fixed in 2673ab68 — added a `delta.Sign() == 0` short-circuit at the top of `updateBalanceMap` so zero deltas are never inserted. Also added a test (`TestUpdateBalanceMap_ZeroDeltaDoesNotInsert`) to cover this edge case.

File: processors/token_transfer/verify_events.go [hunk 2]
```diff
@@ -246,10 +274,19 @@ func VerifyEvents(ledger xdr.LedgerCloseMeta, passphrase string, readFromUnified
...
 		txChangesMap := findBalanceDeltasFromChanges(changes)
 
-		if diff := cmp.Diff(txEventsMap, txChangesMap); diff != "" {
+		bigIntComparer := cmp.Comparer(func(a, b *big.Int) bool {
+			if a == nil || b == nil {
+				return a == b
+			}
+			return a.Cmp(b) == 0
+		})
+		if diff := cmp.Diff(txEventsMap, txChangesMap, bigIntComparer); diff != "" {
```
[the remark this answers is not in the document]
Comment (karthikiyer56) [reply processors/token_transfer/verify_events.go]: Agreed.
 moved `bigIntComparer` above the loop in 2673ab68 — it's stateless so no reason to reallocate per transaction.

**Review by Shaptic — APPROVED** [review]
